### PR TITLE
pkg/flashdb: bump to 2.0.0

### PR DESCRIPTION
**Pull request by benpicco**
### Testing procedure

The tests are still working:

<details><summary>pkg/flashdb_vfs</summary>

```
main(): This is RIOT! (Version: 2023.10-devel-97-gc44a8-pkg/flashdb-2.0)
==================== kvdb_basic_sample ====================
get the 'boot_count' value is 2
set the 'boot_count' value to 3
===========================================================
================= kvdb_type_string_sample =================
create the 'temp' string KV, value is: 36C
get the 'temp' value is: 36C
set 'temp' value to 38C
delete the 'temp' finish
===========================================================
================== kvdb_type_blob_sample ==================
create the 'temp' blob KV, value is: 36
get the 'temp' value is: 36
set 'temp' value to 38
delete the 'temp' finish
===========================================================
======================= tsdb_sample =======================
append the new status.temp (36) and status.humi (85)
append the new status.temp (38) and status.humi (90)
[query_cb] queried a TSL: time: 1691422347, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691422348, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691422832, temp: 36, humi: 85
query count is: 0
set the TSL (time 1691422347) status from 3 to 3
set the TSL (time 1691422348) status from 3 to 3
set the TSL (time 1691422832) status from 2 to 3
===========================================================
```
</details>

<details><summary>pkg/flashdb_mtd</summary>

```
main(): This is RIOT! (Version: 2023.10-devel-97-gc44a8-pkg/flashdb-2.0)
[I/FAL] Flash Abstraction Layer (V0.5.99) initialize success.
==================== kvdb_basic_sample ====================
get the 'boot_count' value is 8
set the 'boot_count' value to 9
===========================================================
================= kvdb_type_string_sample =================
create the 'temp' string KV, value is: 36C
get the 'temp' value is: 36C
set 'temp' value to 38C
delete the 'temp' finish
===========================================================
================== kvdb_type_blob_sample ==================
create the 'temp' blob KV, value is: 36
get the 'temp' value is: 36
set 'temp' value to 38
delete the 'temp' finish
===========================================================
======================= tsdb_sample =======================
append the new status.temp (36) and status.humi (85)
append the new status.temp (38) and status.humi (90)
[query_cb] queried a TSL: time: 1691421584, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691421587, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691421588, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691421589, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691421590, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691421591, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691421895, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691421909, temp: 36, humi: 85
[query_cb] queried a TSL: time: 1691422862, temp: 36, humi: 85
query count is: 0
set the TSL (time 1691421584) status from 3 to 3
set the TSL (time 1691421587) status from 3 to 3
set the TSL (time 1691421588) status from 3 to 3
set the TSL (time 1691421589) status from 3 to 3
set the TSL (time 1691421590) status from 3 to 3
set the TSL (time 1691421591) status from 3 to 3
set the TSL (time 1691421895) status from 3 to 3
set the TSL (time 1691421909) status from 3 to 3
set the TSL (time 1691422862) status from 2 to 3
===========================================================
```
</details>

### CODE REVIEW

File: pkg/flashdb/Makefile [hunk 1]
```diff
@@ -11,6 +11,8 @@ CFLAGS += -Wno-cast-align
 CFLAGS += -Wno-unused
 CFLAGS += -Wno-unused-parameter
 CFLAGS += -Wno-format
+# can be dropped if #226 is merged
```
Comment (kfessel) [inline pkg/flashdb/Makefile:14]: upstream got it

File: pkg/flashdb/patches/0002-port-fal-add-option-to-use-native-assert.patch [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+diff --git a/port/fal/inc/fal_def.h b/port/fal/inc/fal_def.h
+index f558598..de8aada 100644
+--- a/port/fal/inc/fal_def.h
++++ b/port/fal/inc/fal_def.h
+@@ -13,6 +13,9 @@
+ 
+ #include <stdint.h>
+ #include <stdio.h>
++#ifdef FDB_USING_NATIVE_ASSERT
```
Comment (kfessel) [inline pkg/flashdb/patches/0002-port-fal-add-option-to-use-native-assert.patch:18]: upstream got it

**Review by kfessel — COMMENTED** [review]
your PRs are merged into upstream

**Comment by benpicco** [comment]
bors merge

**Pull request merged**